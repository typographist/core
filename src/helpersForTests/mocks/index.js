const breakpointsModel = require('./breakpointsModel');
const userConfig = require('./userConfig');

module.exports = {
  breakpointsModel,
  userConfig,
};
