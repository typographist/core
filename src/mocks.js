// @flow

type Breakpoint = {|
  base: number | number[],
  lineHeight: number,
  name: string,
  ratio: number,
  root: number,
  value: string,
|};

export type Breakpoints = Breakpoint[];

export const breakpoints: Breakpoints = [
  {
    base: [16, 33],
    lineHeight: 1.5,
    name: 'default',
    ratio: 1.1342044031840226,
    root: 12,
    value: '0px',
  },
  {
    base: 17,
    lineHeight: 1.5,
    name: 'tablet',
    ratio: 1.1258335640240535,
    root: 13,
    value: '640px',
  },
  {
    base: 18,
    lineHeight: 1.7,
    name: 'desktop',
    ratio: 1.333,
    root: 16,
    value: '1024px',
  },
  {
    base: 20,
    lineHeight: 1.7,
    name: 'lgDesktop',
    ratio: 1.333,
    root: 17,
    value: '1200px',
  },
  {
    base: 22,
    lineHeight: 1.7,
    name: 'xlDesktop',
    ratio: 1.333,
    root: 19,
    value: '1600px',
  },
];

export const userConfig = {
  base: ['1em', '2em'],
  lineHeight: 1.5,
  ratio: '45px at 6',
  tablet: {
    value: '40em',
    base: '17px',
  },
  desktop: {
    value: '64em',
    base: '18px',
    lineHeight: 1.7,
    ratio: 1.333,
  },
  lgDesktop: {
    value: '75em',
    base: '20px',
  },
  xlDesktop: {
    value: '100em',
    base: '22px',
  },
};
