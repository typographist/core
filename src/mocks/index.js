export { breakpoints, breakpointsModel } from './breakpoints';
export { userConfig, invalidUserConfig } from './user-configs';
