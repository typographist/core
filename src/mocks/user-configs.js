export const userConfig = {
  base: ['16px', '32px'],
  lineHeight: 1.5,
  ratio: '45px at 6',
  tablet: {
    breakpoint: '768px',
    base: '17px',
  },
  desktop: {
    breakpoint: '992px',
    base: '18px',
    lineHeight: 1.7,
    ratio: 1.333,
  },
  lgDesktop: {
    breakpoint: '1200px',
    base: '20px',
  },
  xlDesktop: {
    breakpoint: '1600px',
    base: '22px',
  },
};

export const invalidUserConfig = {
  base: ['1rem', '2em'],
  lineHeight: 1.5,
  ratio: '45 at 6',
  tablet: {
    base: '17px',
  },
  desktop: {
    breakpoint: '64rem',
    base: '18px',
    lineHeight: '1.7',
    ratio: 1.333,
  },
  lgDesktop: {
    breakpoint: '75rem',
    base: '20px',
  },
  xlDesktop: {
    breakpoint: '100rem',
  },
};
