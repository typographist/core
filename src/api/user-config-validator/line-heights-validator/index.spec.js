import { userConfig as config } from '../../../mocks';
import { getLineHeights, isNumberWithException, validateFields } from '.';

describe('validateUserConfig', () => {
  describe('LineHeightsValidator', () => {
    describe('isNumber', () => {
      it('if lineHeight is valid', () => {
        expect(isNumberWithException(1.4)).toBe(true);
      });

      it('if lineHeight is not a number', () => {
        expect(isNumberWithException).toThrowErrorMatchingSnapshot();
      });
    });

    describe('getLineHeights', () => {
      it('returs all values of lineHeight from user config', () => {
        expect(getLineHeights(config)).toEqual([1.5, 1.7]);
      });
    });

    describe('validateFields', () => {
      it('all lineHeights are valid', () => {
        expect(validateFields(config)).toBe(true);
      });
    });
  });
});
