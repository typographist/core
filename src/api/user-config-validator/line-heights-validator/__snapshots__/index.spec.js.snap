// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validateUserConfig LineHeightsValidator isNumber if lineHeight is not a number 1`] = `"Typographist >>> Check your user config. lineHeight:  must be a number. For example lineHeight: 1.5"`;
